src/west/commands/configuration.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''West config commands'''
+
+import argparse
+import configparser
+from west import config
+from west.config import ConfigFile
+from west.commands import WestCommand
+
+CONFIG_DESCRIPTION = '''\
+West configuration file handling.
+
+Command to assist user in getting or setting west project setting
+'.west/config' or west user configuration file '~/.westconfig'.
+
+To get a value for <name>, type
+    west config <name>
+
+To set a value for <name>, type
+    west config <name> <value>
+'''
+
+
+class Configuration(WestCommand):
+    def __init__(self):
+        super().__init__(
+            'config',
+            'get or set configuration settings in west config files',
+            CONFIG_DESCRIPTION)
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=self.description)
+
+        parser.add_argument(
+            'name',
+            help='name is section and key seperated by a dot')
+
+        parser.add_argument(
+            'value',
+            nargs='?',
+            help='Value to set in config file for the given name')
+
+        mx_group = parser.add_mutually_exclusive_group()
+        mx_group.add_argument('--global',
+                              dest='conf_global',
+                              action='store_true',
+                              help='Use global ~/.westconfig for values')
+        mx_group.add_argument('--local',
+                              action='store_true',
+                              help='''Use project specific .west/config for
+                              values''')
+        mx_group.add_argument('--system',
+                              action='store_true',
+                              help='''Use system specific west config for
+                              values''')
+        return parser
+
+    def do_run(self, args, user_args):
+        configfile = ConfigFile.ALL
+        config_settings = configparser.ConfigParser()
+        if args.conf_global:
+            configfile = ConfigFile.GLOBAL
+        elif args.local:
```

Comment:
philosophical question: should this be `if` instead of `elif` since argparse guarantees mutual exclusion?

Reply:
as you point out, they are mutual exclusive, which exactly means we can do the `elif` here.

My preference for choosing `elif`, is that  when `args.conf_global` is set, then `elif` will never be tested.
With
```
if args.conf_global:
    ....
if args.local:
    ....
```
both will always be tested.

I'm happy with both constructs, so let me know if you still prefer to have it changed.

Reply:
I still this the mutual exclusive group is nice here.
Exactly WWGD.

Will take a look at the `action='store_const'` cause that seems to be able to remove the `if ... elif` :)

Reply:
fixed